File: src/lessons/welcome/loopwhile/BaggleSeekerEntity.scala
package lessons.welcome.loopwhile;

import jlm.universe.bugglequest.SimpleBuggle;
import jlm.core.model.Game

class ScalaBaggleSeekerEntity extends SimpleBuggle {
	override def forward(i: Int)  { 
		throw new RuntimeException(Game.i18n.tr("I'm sorry Dave, I'm affraid I can't let you use forward with an argument."));
	}

	override def backward(i: Int) {
		throw new RuntimeException(Game.i18n.tr("I'm sorry Dave, I'm affraid I can't let you use backward with an argument."));
	}

	override def run() { 
		/* BEGIN SOLUTION */
		while (!isOverBaggle()) {
			forward();
		}
		/* END SOLUTION */
	}
}

File: src/lessons/welcome/loopwhile/BaggleSeekerEntity.java
package lessons.welcome.loopwhile;

import plm.universe.bugglequest.SimpleBuggle;

public class BaggleSeekerEntity extends SimpleBuggle {
	
	@Override
	public void forward(int i)  { 
		throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use forward with an argument in this exercise. Use a loop instead."));
	}

	@Override
	public void backward(int i) {
		throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use backward with an argument in this exercise. Use a loop instead."));
	}


	@Override
	public void run() { 
		/* BEGIN TEMPLATE */
		/* BEGIN SOLUTION */
		while (!isOverBaggle()) {
			forward();
		}
		/* END SOLUTION */
		/* END TEMPLATE */
	}
}
